app/controllers/providers/pdf_spike_controller.rb

```diff
@@ -0,0 +1,11 @@
+module Providers
+  class PdfSpikeController < ProviderBaseController
+    layout 'pdf'
+
+    def index
+      authorize @legal_aid_application
+      render pdf: 'file_name',
+             show_as_html: params.key?('debug')
+    end
```

Comment:
I think it is worth trying to see if we can make this a little more restful. Specifically, so that:

- `providers/pdf_spikes` renders HTML
- `providers/pdf_spikes.pdf` renders PDF

I think something like this will do it:
```ruby
respond_to do |format|
  format.html # renders "normally" as html
  format.pdf do
    render pdf: 'file_name',
           show_as_html: params.key?(:debug),
           layout: :pdf
  end
end
```
Note that here the layout is specified in the `render` call rather than at the class level - and therefore is only used for the PDF view.

If that works, it will be much easier to add PDF actions to current views, and we won't need separate PDF controllers.

Reply:
Good idea. I updated the code.
With this solution, we need two templates. `index.html.erb` and `index.pdf.erb` but they can both render the same partial